SmartDeviceLink/SDLAudioStreamManager.h

```diff
@@ -0,0 +1,60 @@
+//
+//  SDLBinaryAudioManager.h
+//  SmartDeviceLink-Example
+//
+//  Created by Joel Fischer on 10/24/17.
+//  Copyright © 2017 smartdevicelink. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@class SDLAudioFile;
+@class SDLManager;
+@class SDLStreamingMediaLifecycleManager;
+@protocol SDLStreamingAudioManagerType;
+@protocol SDLAudioStreamManagerDelegate;
+
+NS_ASSUME_NONNULL_BEGIN
+
+extern NSString *const SDLErrorDomainAudioStreamManager;
+
+typedef NS_ENUM(NSInteger, SDLAudioStreamManagerError) {
+    SDLAudioStreamManagerErrorNotConnected = -1,
+    SDLAudioStreamManagerErrorNoQueuedAudio = -2
+};
+
+@interface SDLAudioStreamManager : NSObject
+
+@property (weak, nonatomic) id<SDLAudioStreamManagerDelegate> delegate;
+
+@property (assign, nonatomic, readonly, getter=isPlaying) BOOL playing;
+@property (copy, nonatomic, readonly) NSArray<SDLAudioFile *> *queue;
+
+- (instancetype)init NS_UNAVAILABLE;
+
+- (instancetype)initWithManager:(id<SDLStreamingAudioManagerType>)streamManager NS_DESIGNATED_INITIALIZER;
+
+/**
+ Push a new file URL onto the queue after converting it into the correct PCM format for streaming binary data.
+
+ @note This happens on a global background thread and will provide an error callback using the delegate if the conversion fails.
+
+ @param fileURL File URL to convert
+ */
+- (void)pushWithFileURL:(NSURL *)fileURL;
```

Comment:
You should probably note that the conversion is asynchronous. That is, if the user uploads a bunch of sound files at once, the files will not necessarily play in the order they were uploaded.

Reply:
What do you mean by "uploads?" They should absolutely play in the order in which they were "pushed"

Reply:
I uploaded 3 sound files right after eachother other in this order: [A (large file), B (small file), C (small file)] and it played in this order: [B, A, C]

Reply:
What does the code look like to push and play here? Did you wait for them all to encode? Did you check the queue to see what files were there in what order? Any chance you did any debugging to see why? They're encoded on a serial queue, so one absolutely should not finish before one started before it.

Reply:
I see what is happening. For each file I pushed, I also triggered a `playNextWhenReady` notification at the same time since I wanted them to play right after each other. The delegate is notified when a track finished based on the length of the song and not on when the track actually finished.